lib/pagseguro/authorization.rb

```diff
@@ -17,14 +17,28 @@ class Authorization
 
     # Find an authorization by it's notification code
     def self.find_by_notification_code(code, options = {})
-      request = Request.get("authorizations/notifications/#{code}", options)
+      request = Request.get("authorizations/notifications/#{code}", "v2", options)
       new Response.new(request).serialize
     end
 
     # Find an authorization by it's code
     def self.find_by_code(code, options = {})
-      request = Request.get("authorizations/#{code}", options)
+      request = Request.get("authorizations/#{code}", "v2", options)
       new Response.new(request).serialize
     end
+
+    def self.find_by_date(options)
+      request = Request.get("authorizations", "v2", RequestSerializer.new(options).to_params)
+      response = Response.new(request).serialize_collection
+      Collection.new(response)
```

Comment:
@vitorkusiaki please, take a look at branch 2.2.0-dev to see a possible new approach for collection, as we need to set errors when request fail.